### PR TITLE
5353 bourbon table

### PR DESCRIPTION
**Pull request by transitorytammy**
screen shot 2019-01-09 at 11 14 47 am

### CODE REVIEW

File: addon/tailwind/components/bourbon-table.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+.BourbonTable-rowItem:first-child {
+  color: #000;
+  font-weight: 500;
+}
+
+.BourbonTable-rowItem:last-child * {
+  float: right;
+}
```
Comment (amacdougall) [inline addon/tailwind/components/bourbon-table.css]: What effect is produced by `float: right`-ing all children of the last row? My CSS isn't sharp enough to visualize it.

Comment (transitorytammy) [reply addon/tailwind/components/bourbon-table.css]: It is basically making the last item hug the end of the row.  Same thing as `text-align: right`.  We wanted to create space between the other items and the last item in the row.

File: addon/templates/components/bourbon-table.hbs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  <tr class="BourbonTable-row">
+    {{#each row as |rowItem|}}
+      <td class="BourbonTable-rowItem">
+        {{#if (eq rowItem.type "component")}}
+          {{component rowItem.content.component title=rowItem.content.title classNames=rowItem.content.classNames}}
+        {{else if (eq rowItem.type "string")}}
+            {{rowItem.content}}
+        {{else if (eq rowItem.type "partial")}}
+            {{partial rowItem.content}}
+        {{/if}}
```
[the remark this answers is not in the document]
Comment (transitorytammy) [reply addon/templates/components/bourbon-table.hbs]: I will add an option for the component to the header just like the row has now.  I think that if and when we use more complex components in this table, we still might need to do some tweaking though.

File: tests/dummy/app/controllers/freestyle.js [hunk 3]
```diff
@@ -8,7 +8,63 @@ const { inject } = Ember;
...
+        { content: "new contact activity", type: "string" },
+        { content: "contact", type: "string"},
+        { content: "create activity", type: "string"},
+        {
+          content: {
+            component: 'bourbon-button',
+            title: 'Remove',
+            classNames: "BourbonButton--secondary"
+          },
+          type: "component"
```
[the remark this answers is not in the document]
Comment (transitorytammy) [reply tests/dummy/app/controllers/freestyle.js]: I think the goal would be to have it eventually take in any ember component being passed in from Flabongo.  For example, I was thinking of how we currently have the some of the cells in the response table that is a bit more complex like that.

Why do you think the ability to take in any ember component would be a bad idea?

[the remark this answers is not in the document]
Comment (amacdougall) [reply tests/dummy/app/controllers/freestyle.js]: It wouldn't be a bad idea—it just has the potential to get out of hand. If it does, though, that's the fault of the developer who did it!

**Comment by transitorytammy** [comment]
1 day in January

**Pull request merged**